### ISSUE TITLE
Visemes somehow get stuck with open mouth after CGE modifications made to FX controller

### ISSUE THREAD

**Issue by hakanai** [hai-vr/combo-gesture-expressions-av3#320]
My setup:

![image](https://user-images.githubusercontent.com/43236/133632543-016ae375-02ab-4d74-a988-1787d5fedfb2.png)
![image](https://user-images.githubusercontent.com/43236/133632628-51ef0984-a177-445c-8e6c-6d1d4368cf0a.png)
![image](https://user-images.githubusercontent.com/43236/133632727-f65e42b7-b23f-4321-a8fd-4b7fcc2f047d.png)
(No changes to default on the other three tabs)

Result after Synchronize Animator FX Layers turns out like this:
(Captured here in Unity but discovered in VRChat)

![image](https://user-images.githubusercontent.com/43236/133632278-5a8d4cde-a5bf-4a9e-a249-9691aeb9784b.png)

Actually it starts out with mouth closed at Viseme=0. Then I randomly change the value of Viseme for a while to reproduce the issue and the mouth gradually gets stuck wider and wider. In-game we got it to something quite grotesque.

Deleted the generated `_Hai_` layers from the animator again, everything back to normal, visemes work as expected, but obviously I have no facial expressions anymore.

At this point though I've used the tool for the most valuable thing which is the setup of the animations themselves, and doing a facial gesture controller manually doesn't seem like a very hard thing to do anyway.

Versions in play:
- Unity 2019.4.30f1
- VRCSDK 2019.2.4-SDK1
- ComboGestureExpressions V1.5.2

**Comment by hai-vr** [comment]
Hi,

Thanks for the report, sorry this happened, I have a few questions:

- What Write Defaults setting did you choose for the FX layer of the ComboGesture compiler?
- Are you using a custom Viseme animation controller (i.e. Xiexe's)?
- In general, is your animation controller Write Defaults ON or OFF?

**Comment by hakanai** [comment]
Answers:

1. Follow VrChat Recommendation Write Defaults
2. Yes. Not Xiexe's, I haven't seen what his is doing but it's probably more complex than what I have set up. Mine is literally just 
![image](https://user-images.githubusercontent.com/43236/133749597-f9dfe642-3cc2-414c-9003-9f8b01a5c810.png)
3. Off I _think_. Not 100% confident because I feel like I might have seen something somewhere else called Write Defaults that I don't remember.

**Comment by hai-vr** [comment]
Thank you for the answers, if you have some time, I'd like to I ask about the Viseme controller, could you please take:
- a screenshot of the inspector with the "06ch" state selected
- another screenshot of the inspector with the transition from Any State to "06ch" selected, with the "settings" section expanded
![image](https://user-images.githubusercontent.com/60819407/133828482-0c64726f-43fb-4677-833d-99f56076c6ce.png)

Normally, the issue you are describing "mouth getting stuck" is presumably the animation from one viseme being stack on top of another animation from another viseme. Normally I only see this happening under the following conditions:
- The viseme animation files are non-exhaustively animating visemes that are animated by other animations of the Viseme sublayer
- and in addition, the Viseme sublayer uses Write Defaults ON
- and:
  - there is at least one state in the animator that uses Write Defaults OFF,
  - or, there is at least one transition in the animator other than the Viseme controller that uses Interruption Source set to something other than None
- or there could be another weird case happening

It's hard to tell right now, so I'm unsure what causes it, because with Write Defaults OFF in an animator, I expect the Viseme sublayer to be functional before and after installing CGE, regardless of the Write Defaults setting of CGE, so I'd greatly appreciate more information about it through those screenshots.

**Comment by hakanai** [comment]
I think I'm at the right screen now.

[image: image.png]

**Comment by hakanai** [comment]
Yes, I don't know why that would be off or on at all.
It was taken from a publicly available avatar which had been behaving so
far.

**Comment by hai-vr** [comment]
> 
> I think I'm at the right screen now. [image: image.png]

There is no picture showing up on that message?

**Comment by hakanai** [comment]
Yeah it looks like the inline image in the mail got deleted by GitHub.

https://user-images.githubusercontent.com/43236/134291614-d938fe40-7764-4c61-8a68-17750d30fa28.png

**Comment by hakanai** [comment]
So I was talking with someone else about this and we came to the conclusion that mixing Write Defaults on and off in the same controller is going to cause a bad time. Reading the docs again I see that this is recommended there as well. But I don't get what it does with enough confidence to go and turn it off everywhere when the default is apparently on and other examples I find out there have it on as well. 😨 

So what I'm concluding is that I should have chosen this to be on, even though it says "Unsupported"?

But saying "Unsupported" is a really good way to discourage choosing the option. It worked on me, at least. I guess instead of having scare text, it should have some kind of way to let me figure out for myself which option is correct to choose.
